students/Mintae_K/lesson05/mailroom3.py

```diff
@@ -0,0 +1,80 @@
+# Error Handling function for sending_thank function
+def check_error(f_name, s):
+    while True:
+        amount = input(s.format(f_name))
+        try:
+            checked_amount = float(amount)
+            break
+        except ValueError:
+            print('\nValue entered is not a number. Only number allowed.\n')
+    return checked_amount
+
+
+# Sending a Thank you
+def sending_thank():
+    f_name = input('Please type the Full Name or list> ')
+    while f_name == 'list':
+        print(' ')
+        for x in donors:
+            print(x)
+        print(' ')
+        f_name = input('Please type the Full Name or list> ')
+    if donors.get(f_name):
+        s = 'Existing donor, please type the donation amount from {}> '
+        donors[f_name].append(check_error(f_name, s))
+    else:
+        s = 'New donor, please type the donation amount from {}> '
+        donors[f_name] = [check_error(f_name, s)]
+    print('\n', "Thank you {} for the generous donation!".format(f_name), '\n')
+
+
+# Creating Report
+def create_report():
+    s = '\n, {:<20}| {:<10} | {:<10}| {:<20}'
+    print(s.format('Donor Name', 'Total Given', "Num Gift", "Average Gift"))
+    print('-' * 60)
+    total = [[sum(donors[key]), len(donors[key]), key] for key in donors]
+    total.sort(reverse=True)
+    s = '{:<20}  $ {:<10.2f}  {:^12} $ {:<20.2f}'
+    for i in range(len(total)):
+        print(s.format(total[i][2], total[i][0], total[i][1], total[i][0] / total[i][1]))
+    print(' ')
+
+
+def send_letters():
+    for name in donors:
+        s = "{}.txt".format(name)
+        with open(s, "w+") as f:
+            f.write("Dear {},".format(name))
+            f.write("\n\n     Thank you for your very kind donation of ${:<10.2f}".format(sum(donors[key])))
+            f.write('\n\n     It will be put to very good use.')
+            f.write('\n\n             Sincerely,')
+            f.write('\n                   -The Team')
+        f.closed
```

Comment:
Getting an error when selecting this option - "NameError: name 'key' is not defined"